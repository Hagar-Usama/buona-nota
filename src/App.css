.App {
  text-align: center;
  
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}


  .Ticket {
  width: 80%; 
  height: 80%; 
  margin: auto;
  text-align: center;
  flex:1;
  background-color: rgba(50,50,50,0.7);
   

}

.Container{

display: flex;
flex-direction: row;
flex-wrap: wrap;
flex-flow: column wrap;
justify-content:flex-end;
align-items: flex-end;
align-content: flex-start;


}

.container {
  flex: 1;
  justifyContent: 'center';
  paddingTop: Constants.statusBarHeight;
  backgroundColor: '#ecf0f1';
  padding: 18;
  
}

.ed {

  visibility: visible;
}



.navbar {
  overflow: hidden;
  background-color:rgba(50,50,50,0.1);
  position: fixed;
  width: 100%;
  bottom: 0;
  right: 0;
}


.navbar a:hover {
  background: #ddd;
  position: fixed;
  color: black;
}


.loginBox {
  position: relative;
  top: 50%;
  left: 50%;
  width: 400px;
  padding: 40px 50px 50px 40px;
  transform: translate(-50%, -50%);
  background: rgba(230, 226, 226, 1);
  box-sizing: border-box;
  border-radius: 10px;
}

.blockquote-footer{

  visibility: hidden;
}


.head{

  text-align: center;
  align-content: center;
  font-family: Stencil Std, fantasy;

}

.sticky {
  position: fixed;
  top: 0;
  width: 100%;
}

.hid{
visibility: hidden;

}