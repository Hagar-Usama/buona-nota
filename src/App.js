
import React, { Component } from 'react';
import { Row } from 'reactstrap';
import Ticket from './ticket'
import Head from './head'
import { Button, Card, CardBody, CardText, CardGroup, CardTitle } from 'reactstrap';
import './App.css';
const getStatus = ({status} ) => 
(status ? "Done" : "Complete me!");


class App extends Component {

  constructor(props){
    super(props);
    this.state = {todos: [] , id: 0};
    this.deleteNote = this.deleteNote.bind(this);
    this.handleDelete = this.handleDelete.bind(this);
    this.handleEdit = this.handleEdit.bind(this);
    this.handleAdd = this.handleAdd.bind(this);
    this.incrementId = this.incrementId.bind(this);
  }

  deleteNote(id){

    let filtered = this.state.todos.filter(it => it.id !== id);
    this.setState({todos: filtered});

  }

  handleEdit(id, message){
   console.log("edit from App");
   console.log(id, message)
   let elementsIndex = (this.state.todos.findIndex(element => element.id === id ))
   console.log(elementsIndex)
   let newArray = [...this.state.todos]
   newArray[elementsIndex] = {...newArray[elementsIndex], title: message}
    
    this.setState({
      todos: newArray,
      });
   


  }

 


  getId(){
   
    if (this.state.id === 0){

      //this.setState({id: this.state.todos[2].id})
    this.state.id =  this.state.todos[this.state.todos.length-1].id
    console.log("set id")
    console.log(this.state.id)
    }
    

  }

  incrementId(){

    this.state.id++;
  }
 handleDelete(id){

    console.log("Hello from App");
    this.deleteNote(id)
  }

  handleAdd(message){

    this.getId();
    this.incrementId();
    console.log("Id ")
    console.log(this.state.id)
    console.log("Hello from App");
    let msg = {

      userId: 1,
      id:this.state.id,
      title: message,
      completed: false
    };

    this.addNote(msg)

  

  }

  addNote(taskObj){

    this.setState(
      { todos: [taskObj, ...this.state.todos] });

  }
 
  componentDidMount() {
    document.title = "Buoana Nota"
    fetch('http://jsonplaceholder.typicode.com/todos')
    .then(result => result.json())
    .then((data) => {
      this.setState({ todos: data })
      console.log("todos in mount")
      console.log(this.state.todos)
      let filtered = this.state.todos.filter(it => it.id > 150);
      console.log(filtered)
    })
    .catch(console.log)

    
  }


 render() {

  return (
     <div  >
       
     
     <Head  addMethod={this.handleAdd}/>
     <hr/>
    
     <Row>
      {this.state.todos.map((todo) => (

        <Ticket title={todo.title} status= {getStatus(todo.completed)} id={todo.id}  deleteMethod={this.handleDelete} editMethod={this.handleEdit} addMethond={this.handleAdd}/>

                
      ))}
      
      </Row>
      
     </div>
  );
}

}

export default App;
