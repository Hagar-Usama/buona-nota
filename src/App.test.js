import React from 'react';

import renderer from 'react-test-renderer';
import Ticket from './ticket';


import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import App from './App';
jest.mock('axios');

configure({adapter: new Adapter()});
//Enzyme.configure({ adapter: new Adapter() })

describe('App component', () => {
  it('App: renders correctly', () => {
    const tree = renderer.create(<App />);
    expect(tree).toMatchSnapshot();
  });

  it('test state id', () => {
    const wrapper = shallow(<App/>);
    expect(wrapper.instance().state.id).toBe(0);
  });

  it('test todos', () => {
    const wrapper = shallow(<App/>);
    const instance = wrapper.instance();
    instance.componentDidMount();
    instance.fetchData();
    instance.incrementId();
    const data = instance.state
    console.log({data})
    
  });

});


describe('Ticket component', () => {
  it('Ticket: renders correctly', () => {
    const tree = renderer.create(<Ticket />);
    expect(tree).toMatchSnapshot();
  });


});


test("fake", () => {

  expect(1 + 1).toBe(2);

})

