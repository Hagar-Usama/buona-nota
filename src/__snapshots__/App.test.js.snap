// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App: renders correctly 1`] = `
<div
  className="App"
>
  <container
    className="head"
  >
    <div
      className="navbar"
    >
      <h1
        className="head"
      >
        ToDo or not ToDo!
      </h1>
      <div
        className="navbar input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
          >
            +
          </span>
        </div>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Add a Task"
          type="text"
        />
        <button
          aria-label={null}
          className="btn btn-warning"
          onClick={[Function]}
          type="button"
        >
          Add
        </button>
      </div>
      <br />
    </div>
    <hr />
  </container>
  <div
    className="row"
  />
</div>
`;

exports[`Ticket component Ticket: renders correctly 1`] = `
<div
  className="col-sm-4"
>
   
  <div
    className="Ticket card"
  >
    <div
      className="card-body"
    >
      <button
        aria-label="Cancel"
        className="close"
        onClick={[Function]}
        type="button"
      >
        <span
          aria-hidden={true}
        >
          –
        </span>
      </button>
      <div
        active={true}
        className="card-subtitle"
      />
      <hr />
      <textarea
        className="ed form-control"
        cols={5}
        disabled="disabled"
        onChange={[Function]}
        rows={5}
      />
      <hr />
      <button
        aria-label={null}
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Edit
      </button>
       
      <button
        aria-label={null}
        className="btn btn-danger"
        onClick={[Function]}
        type="button"
      >
        Delete
      </button>
    </div>
  </div>
  <hr />
   
</div>
`;
